### ISSUE TITLE
AttributeError: 'str' object has no attribute 'set_array' when calling kdeplot with legend=True

### ISSUE THREAD

**Issue by nickderobertis** [ResidentMario/geoplot#229]
First, thanks for the great library.

I noticed an issue when trying to use the kdeplot function when also passing legend=True. It works fine when legend is not passed. Here is a code sample which is failing for me:

```python
import pandas as pd
import geopandas
import geoplot

df = pd.DataFrame(
    [
       [ 27.547044 , -82.533896 ],
       [ 27.4853062, -82.5652603],
       [ 27.4782766, -82.5937605],
       [ 27.5018049, -82.6089842],
       [ 27.476826 , -82.5190818],
       [ 27.49207  , -82.5964237],
    ],
    columns=['latitude', 'longitude']
)
gdf = geopandas.GeoDataFrame(
    df, geometry=geopandas.points_from_xy(df.longitude, df.latitude)
)
gdf.crs = 4326
geoplot.kdeplot(gdf, shade=True, legend=True)
```

From that I get the following traceback:

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-29-ba321c5cab0f> in <module>
     17 )
     18 gdf.crs = 4326
---> 19 geoplot.kdeplot(gdf, shade=True, legend=True)

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/geoplot/geoplot.py in kdeplot(df, projection, extent, figsize, ax, clip, shade_lowest, **kwargs)
   1360             return ax
   1361 
-> 1362     plot = KDEPlot(
   1363         df, projection=projection, extent=extent, figsize=figsize, ax=ax, clip=clip,
   1364         shade_lowest=shade_lowest,

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/geoplot/geoplot.py in __init__(self, df, **kwargs)
   1336                 verify_input=False
   1337             )
-> 1338             self.paint_legend(supports_hue=True, supports_scale=False, verify_input=False)
   1339             self.paint_clip()
   1340 

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/geoplot/geoplot.py in paint_legend(self, supports_hue, supports_scale, verify_input, scale_multiplier)
    354                     )
    355 
--> 356                 self.cmap.set_array(self.hue)
    357                 try:
    358                     plt.gcf().colorbar(self.cmap, ax=self.ax, **legend_kwargs)

AttributeError: 'str' object has no attribute 'set_array'
```

Here are the versions of Python and relevant packages:

```
Python: 3.8.5
OS: Ubuntu 20.04
geoplot==0.4.1
geopandas==0.8.1
seaborn==0.11.0
matplotlib==3.3.2
pandas==0.25.3
Cartopy==0.18.0
descartes==1.1.0
contextily==1.0.1
mapclassify==2.3.0
```

**Comment by ResidentMario** [comment]
Hi, thanks for reporting this issue! This looks like a legitimate bug in the implementation due to improper handling of the `legend` parameter. Curiously, when I update the code to pass it through as a `kwarg`, the legend still doesn't display.

For now, setting `legend=True` should fix your issue.

**Comment by nickderobertis** [comment]
@ResidentMario Thanks for looking into this so quickly! I definitely appreciate your work on this library because this KDE plot handles my use case perfectly and is very convenient.

**Comment by ResidentMario** [comment]
Should be solved in #271 and #272.
